frontend/src/metabase/querying/filters/utils/dates.ts

```diff
@@ -150,7 +181,48 @@ function getExcludeFilterClause(
   });
 }
 
-export function getPickerOperators(
+function getMonthYearFilterClause(
+  column: Lib.ColumnMetadata,
+  value: MonthYearPickerValue,
+): Lib.ExpressionClause {
+  const startOfMonth = dayjs()
+    .year(value.year)
+    .month(value.month - 1)
+    .startOf("month")
+    .toDate();
+  const endOfMonth = dayjs(startOfMonth).endOf("month").startOf("day").toDate();
```

Comment:
What's the point of `.startOf("day")`?
It does not make a difference now and feels potentially incorrect if we ever wanted to have greater granularity here.

This comment also applies to `getQuarterYearFilterClause`.

```suggestion
  const endOfMonth = dayjs(startOfMonth).endOf("month").toDate();
```

Reply:
With `startOf("day")` -> Jan 1 - 31, 2024`.
Without - `Jan 1, 12:00 AM – Jan 31, 2025, 11:59:59.999 PM`, which is a) incorrect for dates with higher precision than the browser supports, b) looks odd.